src/swap.swap/Swap.js

```diff
@@ -47,6 +47,20 @@ class Swap {
     }
 
     this.flow = new Flow(this)
+
+    const swap = this;
+
+    // Change destination address on run time
+    this.room.on('set destination buy address', function (data) {
```

Comment:
```suggestion
    this.room.on('set destination buy address', (data) => {
```

тогда можно не делать const swap = this и просто использовать this везде, стрелочные функции его не перезаписывают